coffee_lib/src/types/mod.rs

```diff
@@ -182,13 +208,52 @@ pub mod response {
         pub status: Vec<NurseStatus>,
     }
 
+    impl CoffeeNurse {
+        pub fn is_sane(&self) -> bool {
+            self.status.is_empty()
+        }
+
+        pub fn organize(&mut self) {
+            // For every action taken by the nurse, we want to
+            // have 1 entry with the list of repositories affected.
+            let mut new_status: Vec<NurseStatus> = vec![];
+            let repositories_locally_removed: Vec<String> = self
+                .status
+                .iter()
+                .filter_map(|status| match status {
+                    NurseStatus::RepositoryLocallyRemoved(repos) => Some(repos),
+                    _ => None,
+                })
+                .flatten()
+                .cloned()
+                .collect();
```

Comment:
This is a mess :)

Reply:
true
I couldn't wrap my head around how the values of the vector are `Enum` and also contain a vector of Strings
This bad design from my side

Now that I think about it, I believe this should be just a vector

Reply:
the confusing part is the chaining of cal. You should split the call to make the code readble